cuda/solver/lower_trs_kernels.cu

```diff
@@ -53,10 +58,124 @@ namespace cuda {
 namespace lower_trs {
 
 
+#if (defined(CUDA_VERSION) && (CUDA_VERSION > 9010))
+struct cusp_csrsm2_data {
+    int algorithm;
+    csrsm2Info_t solve_info;
+    cusparseSolvePolicy_t policy;
+    cusparseMatDescr_t factor_descr;
+    size_t factor_work_size;
+    void *factor_work_vec;
+};
+static cusp_csrsm2_data cusp_csrsm2_data{};
+
+#elif (defined(CUDA_VERSION) && (CUDA_VERSION < 9020))
+struct cusp_csrsm_data {
+    cusparseSolveAnalysisInfo_t solve_info;
+    cusparseMatDescr_t factor_descr;
+};
+static cusp_csrsm_data cusp_csrsm_data{};
+#endif
+
+
+void clear(std::shared_ptr<const CudaExecutor> exec)
+{
+#if (defined(CUDA_VERSION) && (CUDA_VERSION > 9010))
```

Comment:
Why are you doing `CUDA_VERSION < 9020` and `CUDA_VERSION > 9010`? What about the numbers between `9010` and `9020`? I think it would be a lot easier to understand (and be future proof in case CUDA adds these numbers) to do:
`CUDA_VERSION < 9020` and
`CUDA_VERSION >= 9020`.

This applies to all of the `#if`s in this file.